File: app/src/main/java/com/adapter/EditaItemAdapter.java
package com.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.R;
import com.entity.DayEntity;
import com.lidroid.xutils.util.LogUtils;

import java.util.List;

/**
 * Created by liguoying on 2017/9/22.
 */

public class EditaItemAdapter extends RecyclerView.Adapter<EditaItemAdapter.EditaItemAdapterViewHolder> {
    private Context mContext;
    private List<DayEntity.ContentBean.DateBean> mDateBeans;

    private int mFirstPosition;
    private IItemSecondItemClick mIItemSecondItemClick;
    private boolean editAble;

    public EditaItemAdapter(Context context, List<DayEntity.ContentBean.DateBean> dateBeans, int position, boolean editAble) {
        mContext = context;
        mDateBeans = dateBeans;
        this.mFirstPosition = position;
        this.editAble = editAble;
    }

    @Override
    public EditaItemAdapterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new EditaItemAdapterViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_sigle_menu, parent, false));
    }

    @Override
    public void onBindViewHolder(final EditaItemAdapterViewHolder holder, final int position) {
        if (!editAble) {
            holder.mCutOrAdd.setVisibility(View.INVISIBLE);
        } else {
            holder.mCutOrAdd.setVisibility(View.VISIBLE);
        }
        holder.logo.setImageResource(R.mipmap.ic_category_0);
        holder.mCutOrAdd.setImageResource(R.mipmap.icon_add_menu);
        holder.name.setText(mDateBeans.get(position).getImage());
        holder.mRelativeLayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                mIItemSecondItemClick.secondItemClick(position, mFirstPosition);
                LogUtils.e(position + "   " + mFirstPosition);
                holder.mCutOrAdd.setImageResource(R.mipmap.icon_selected_yes);
            }
        });
    }

    @Override
    public int getItemCount() {
        return mDateBeans.size();
    }

    class EditaItemAdapterViewHolder extends RecyclerView.ViewHolder {

        private ImageView logo;
        private TextView name;
        private ImageView mCutOrAdd;
        private RelativeLayout mRelativeLayout;

        public EditaItemAdapterViewHolder(View itemView) {
            super(itemView);
            logo = itemView.findViewById(R.id.iv_image_edite_sigle);
            name = itemView.findViewById(R.id.tv_sigle_menu_name);
            mCutOrAdd = itemView.findViewById(R.id.iv_cancler_or_add);
            mRelativeLayout = itemView.findViewById(R.id.ll_item_second);
        }
    }

    public interface IItemSecondItemClick {
        void secondItemClick(int position, int firstPosition);
    }

    public void setSecondItemClick(IItemSecondItemClick mIItemSecondItemClick) {
        this.mIItemSecondItemClick = mIItemSecondItemClick;
    }
}

File: app/src/main/java/com/adapter/EditaItemAdapter.kt
package com.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import com.androidwebviewdemo.mddemo.R
import com.entity.DayEntity
import com.lidroid.xutils.util.LogUtils

/**
 * Created by liguoying on 2017/9/22.
 */

class EditaItemAdapter(val mContext: Context, val mDateBeans: List<DayEntity.ContentBean.DateBean>, val mFirstPosition: Int, val editAble: Boolean) : RecyclerView.Adapter<EditaItemAdapter.EditaItemAdapterViewHolder>() {
    private var mIItemSecondItemClick: IItemSecondItemClick? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EditaItemAdapterViewHolder {
        return EditaItemAdapterViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_sigle_menu, parent, false))
    }

    override fun onBindViewHolder(holder: EditaItemAdapterViewHolder, position: Int) {
        if (!editAble) {
            holder.mCutOrAdd.visibility = View.INVISIBLE
        } else {
            holder.mCutOrAdd.visibility = View.VISIBLE
        }
        holder.logo.setImageResource(R.mipmap.ic_category_0)
        if (mDateBeans[position].status == 1) {
            holder.mCutOrAdd.setImageResource(R.mipmap.icon_add_menu)
        } else {
            holder.mCutOrAdd.setImageResource(R.mipmap.icon_selected_yes)

        }
        holder.name.text = mDateBeans[position].image
        holder.mRelativeLayout.setOnClickListener {
            if (mDateBeans[position].status == 1) {
                holder.mCutOrAdd.setImageResource(R.mipmap.icon_selected_yes)
                mIItemSecondItemClick!!.secondItemClick(position, mFirstPosition)
            }
            LogUtils.e(position.toString() + "   " + mFirstPosition)
        }
    }

    override fun getItemCount(): Int {
        return mDateBeans.size
    }

    inner class EditaItemAdapterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val logo: ImageView
        val name: TextView
        val mCutOrAdd: ImageView
        val mRelativeLayout: RelativeLayout

        init {
            logo = itemView.findViewById<ImageView>(R.id.iv_image_edite_sigle)
            name = itemView.findViewById<TextView>(R.id.tv_sigle_menu_name)
            mCutOrAdd = itemView.findViewById<ImageView>(R.id.iv_cancler_or_add)
            mRelativeLayout = itemView.findViewById<RelativeLayout>(R.id.ll_item_second)
        }
    }

    interface IItemSecondItemClick {
        fun secondItemClick(position: Int, firstPosition: Int)
    }

    fun setSecondItemClick(mIItemSecondItemClick: IItemSecondItemClick) {
        this.mIItemSecondItemClick = mIItemSecondItemClick
    }
}
